registry/client/repository.go

```diff
@@ -604,6 +607,25 @@ func (bs *blobs) Resume(ctx context.Context, id string) (distribution.BlobWriter
 	panic("not implemented")
 }
 
+func (bs *blobs) Mount(ctx context.Context, sourceRepo distribution.Repository, dgst digest.Digest) (distribution.Descriptor, error) {
```

Comment:
@aaronlehmann Does this impose too much on the client to have a fully prepared repository when only the name is required?